<launch> 
	<arg name="use_bow" default="true" />
    <arg name="use_port" default="true" />
    <arg name="use_starboard" default="true" />
    <arg name="use_transom" default="true" />
    <arg name="bow_camera_ip" default="192.168.0.100" />
    <arg name="starboard_camera_ip" default="192.168.0.105" />
    <arg name="port_camera_ip" default="192.168.0.103" />
    <arg name="transom_camera_ip" default="192.168.0.104" />

    <arg name="image_width" default="640" />
    <arg name="image_height" default="480" />
    <arg name="enable_theora" default="1" />

    <arg name="debug_mode_enable" value="0"/> 

    <master auto="start" />

    <group ns="fixed_cam" if="$(arg use_bow)">
        <node pkg="robotx_vision" type="axis.py" name="camera_node" output="screen">
            <param name="hostname" value="$(arg bow_camera_ip)" />
            <param name="username" value="root" />
            <param name="password" value="pass"/>
            <param name="width" value="$(arg image_width)" />
            <param name="height" value="$(arg image_height)" />
            <param name="frame_id" value="camera"/>
            <param name="camera_info_url" type="string" value="package://robotx_vision/camera_info/bow_left_camera.yaml"/>
        </node>

        <node pkg="image_transport" type="republish" name="republish"
            args="compressed" if="$(arg enable_theora)">
            <remap from="in" to="image_raw" />
            <remap from="out" to="image_raw_out" />
        </node>
        <node pkg="image_view" type="image_view" name="image_view" 
            if="$(arg debug_mode_enable)"> 
            <param name="image_transport" value="compressed" />
            <remap from="image" to="image_rect_color" />
        </node>

        <node name="image_proc" pkg="image_proc" type="image_proc" output="screen">
            <remap from="image_raw" to="image_raw_out"/>
            <remap from="camera_info" to="camera_info"/>
            <param name="queue_size" value="20"/>
        </node>
    </group>
    
    <group ns="starboard_cam" if="$(arg use_starboard)">
        <node pkg="robotx_vision" type="axis.py" name="camera_node" output="screen">
            <param name="hostname" value="$(arg starboard_camera_ip)" />
            <param name="username" value="root" />
            <param name="password" value="pass"/>
            <param name="width" value="$(arg image_width)" />
            <param name="height" value="$(arg image_height)" />
            <param name="frame_id" value="camera"/>
            <param name="camera_info_url" type="string" value="package://robotx_vision/camera_info/starboard_camera.yaml"/>
        </node>

        <node pkg="image_transport" type="republish" name="republish"
            args="compressed" if="$(arg enable_theora)">
            <remap from="in" to="image_raw" />
            <remap from="out" to="image_raw_out" />
        </node>
        <node pkg="image_view" type="image_view" name="image_view" 
            if="$(arg debug_mode_enable)"> 
            <param name="image_transport" value="compressed" />
            <remap from="image" to="image_rect_color" />
        </node>

        <node name="image_proc" pkg="image_proc" type="image_proc" output="screen">
            <remap from="image_raw" to="image_raw_out"/>
            <remap from="camera_info" to="camera_info"/>
            <param name="queue_size" value="20"/>
        </node>
    </group>

    <group ns="port_cam" if="$(arg use_port)">
        <node pkg="robotx_vision" type="axis.py" name="camera_node" output="screen">
            <param name="hostname" value="$(arg port_camera_ip)" />
            <param name="username" value="root" />
            <param name="password" value="pass"/>
            <param name="width" value="$(arg image_width)" />
            <param name="height" value="$(arg image_height)" />
            <param name="frame_id" value="camera"/>
            <param name="camera_info_url" type="string" value="package://robotx_vision/camera_info/port_camera.yaml"/>
        </node>

        <node pkg="image_transport" type="republish" name="republish"
            args="compressed" if="$(arg enable_theora)">
            <remap from="in" to="image_raw" />
            <remap from="out" to="image_raw_out" />
        </node>
        <node pkg="image_view" type="image_view" name="image_view" 
            if="$(arg debug_mode_enable)"> 
            <param name="image_transport" value="compressed" />
            <remap from="image" to="image_rect_color" />
        </node>

        <node name="image_proc" pkg="image_proc" type="image_proc" output="screen">
            <remap from="image_raw" to="image_raw_out"/>
            <remap from="camera_info" to="camera_info"/>
            <param name="queue_size" value="20"/>
        </node>
    </group>

    <group ns="transom_cam" if="$(arg use_transom)">
        <node pkg="robotx_vision" type="axis.py" name="camera_node" output="screen">
            <param name="hostname" value="$(arg transom_camera_ip)" />
            <param name="username" value="root" />
            <param name="password" value="pass"/>
            <param name="width" value="$(arg image_width)" />
            <param name="height" value="$(arg image_height)" />
            <param name="frame_id" value="camera"/>
            <param name="camera_info_url" type="string" value="package://robotx_vision/camera_info/transom_camera.yaml"/>
        </node>

        <node pkg="image_transport" type="republish" name="republish"
            args="compressed" if="$(arg enable_theora)">
            <remap from="in" to="image_raw" />
            <remap from="out" to="image_raw_out" />
        </node>

        <node pkg="image_view" type="image_view" name="image_view" 
            if="$(arg debug_mode_enable)"> 
            <param name="image_transport" value="compressed" />
            <remap from="image" to="image_rect_color" />
        </node>

        <node name="image_proc" pkg="image_proc" type="image_proc" output="screen">
            <remap from="image_raw" to="image_raw_out"/>
            <remap from="camera_info" to="camera_info"/>
            <param name="queue_size" value="20"/>
        </node>
    </group>

    <node pkg="tf" type="static_transform_publisher" name="my_tf" args="0 0 0 0 0 0 /map /axis_camera 100"/>
</launch>
